lib/src/main/kotlin/io/wavebeans/lib/stream/window/WindowFunction.kt

```diff
@@ -0,0 +1,66 @@
+package io.wavebeans.lib.stream.window
+
+import io.wavebeans.lib.*
+import io.wavebeans.lib.stream.map
+import kotlin.math.PI
+import kotlin.math.abs
+import kotlin.math.cos
+
+class MapWindowFn<T : Any>(initParameters: FnInitParameters) : Fn<Window<T>, Window<T>>(initParameters) {
+
+    constructor(windowFunction: Fn<Pair<Int, Int>, T>, multiplyFn: Fn<Pair<T, T>, T>) : this(FnInitParameters()
```

Comment:
add KDocs:
* `windowFunction` -- function to be used to generate the values. Has two values: 1. the current index to generate for and  2. overall amount of sample-entities will be asked to generate.
* `multiplyFn` -- function to be used to multiply the corresponding sample-entities while applying the window function.